### PR TITLE
Add grouping to Select from Project

### PR DESCRIPTION
**Pull request by jhadvig**
@spadgett PTAL

Closes https://github.com/openshift/origin-web-console/issues/2332

### CODE REVIEW

File: app/scripts/directives/processTemplateDialog.js [hunk 1]
```diff
@@ -208,6 +208,14 @@
       }));
     };
 
+    // TODO: Make this as a reusable component since we are using similar method also in origin-web-catalog
```
Comment (spadgett) [inline app/scripts/directives/processTemplateDialog.js]: I think the TODO is really to reuse the select-project component. We'd need to update the component to optionally disable creating new projects, however. Do you mind updating the comment?

**Comment by jhadvig** [comment]
@spadgett comment addressed

**Comment by spadgett** [comment]
WebDriver flake #1684

/retest

**Pull request merged**